src/hooks/UseDynamicHighilghtedLink.js

```diff
@@ -0,0 +1,140 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+/**
+ * Custom hook to extract the logic behind the highlighted link in the AutoScrollableMenu
+ *
+ * This hook allow to determine which link should be highlighted based on two inputs.
+ *
+ *  - The first input is the menu link on which we clicked when we click on one.
+ *  - The second input is the highest object from the `elemRefs` list visible inside the `containerRef` element.
```

Comment:
I thought the second input was a list??

Reply:
Yes, you are right, the second element is indeed a list, I think my comment is not clear enough. I will try to reformulate it.